### PR TITLE
.*: refine `executeQuery`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #5900 

Problem Summary:

use `Flash::executeDAG` to replace `Interpreters::executeQueryImpl`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/executeQuery.cpp [hunk 1]
```diff
@@ -12,27 +12,96 @@
...
+        if (dag_context.is_mpp_task)
+        {
+            /// for MPPTask, process list entry is created in MPPTask::prepare()
+            RUNTIME_ASSERT(dag_context.getProcessListEntry() != nullptr, "process list entry for MPP task must not be nullptr");
+            res.process_list_entry = dag_context.getProcessListEntry();
+        }
+        else
+        {
+            RUNTIME_ASSERT(dag_context.getProcessListEntry() == nullptr, "process list entry for non-MPP must be nullptr");
+            res.process_list_entry = setProcessListElement(
```
Comment (windtalker) [inline dbms/src/Flash/executeQuery.cpp:77]: Why `setProcessListElement` after `interpreter->execute()`

Comment (windtalker) [inline dbms/src/Flash/executeQuery.cpp:77]: Why `setProcessListElement` after `interpreter->execute()`

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp:77]: I think order doesn't matter here, just want to reduce one variable process_list_entry
But I can keep the same order as the original implementation.

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp:77]: o.. may be relevant, such as starting an asynchronous read task in `interpreter->execute()`...

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp]: fixed.

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp]: fixed.

File: dbms/src/Flash/executeQuery.cpp [hunk 2]
```diff
@@ -12,27 +12,103 @@
...
+    {
+        RUNTIME_ASSERT(dag_context.getProcessListEntry() == nullptr, "process list entry for non-MPP must be nullptr");
+        return setProcessListElement(
+            context,
+            dag_context.dummy_query_string,
+            dag_context.dummy_ast.get());
+    }
+}
+
+BlockIO executeDAG(IQuerySource & dag, Context & context, bool internal)
```
Comment (windtalker) [inline dbms/src/Flash/executeQuery.cpp:67]: We can remove the `IQuerySource` related code since we don't reuse `executeQuery`?

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp:67]: Yes, but here is some relevant code that needs to be cleaned up.
We can remove the IQuerySource in the next pr.

Comment (SeaRise) [reply dbms/src/Flash/executeQuery.cpp:67]: In particular, the code related to the `DAGQuerySource`, perhaps we could remove the `IQuerySource` along with the `DAGQueryBlock`.

Comment (windtalker) [reply dbms/src/Flash/executeQuery.cpp:67]: OK

**Review by windtalker — APPROVED** [review]
LGTM

**Review by bestwoody — APPROVED** [review]

**Pull request merged**